### PR TITLE
PP-5081 add validation for prefilled cardholder details

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- add new Object (PrefilledCardHolderDetails)
- add jackson validations
- add relevant tests

with @nimalank7

## How to test

- How should it be reviewed? These are 'vanilla' length validations

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/charge/model/PrefilledCardHolderDetails.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.hibernate.validator.constraints.Length;
+import uk.gov.pay.connector.common.model.domain.Address;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class PrefilledCardHolderDetails {
+    
+    @Length(max = 20, message = "Field [cardholder_name] can have a size between 0 and 255")
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/charge/model/PrefilledCardHolderDetails.java]: Should the max be 255 here ?

File: src/main/java/uk/gov/pay/connector/charge/model/ChargeCreateRequest.java [hunk 2]
```diff
@@ -92,6 +99,10 @@ public boolean isDelayedCapture() {
     public SupportedLanguage getLanguage() {
         return language;
     }
+    
+    public PrefilledCardHolderDetails getPrefilledCardHolderDetails() {
+        return prefilledCardHolderDetails;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/charge/model/ChargeCreateRequest.java]: Do we need  Optional<> pattern here ? Probably makes it easy to use Optional when updating database or anywhere it is used

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiCreateResourceITest.java [hunk 3]
```diff
@@ -323,6 +331,58 @@ public void cannotMakeChargeForMissingFields() {
                         "Field [amount] can be between 1 and 10_000_000"
                 ));
     }
+    
+    @Test
+    public void shouldReturn400WhenCardHolderNameIsLongerThan255() {
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiCreateResourceITest.java]: Given that `max = 20` is not detected, we may need a test for happy path for maximum length something like `shouldReturn2xxWhenCardHolderNameIs255`

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiCreateResourceITest.java [hunk 4]
```diff
@@ -323,6 +331,58 @@ public void cannotMakeChargeForMissingFields() {
...
+        connectorRestApiClient.postCreateCharge(postBody)
+                .statusCode(BAD_REQUEST.getStatusCode())
+                .contentType(JSON)
+                .header("Location", is(nullValue()))
+                .body(JSON_CHARGE_KEY, is(nullValue()))
+                .body(JSON_MESSAGE_KEY, containsInAnyOrder("Field [cardholder_name] can have a size between 0 and 255"));
+    }
+
+    @Test
+    public void shouldReturn400WhenAddressFieldsAreLongerThanMaximum() {
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiCreateResourceITest.java]: Similar to above,  a test case up to max length ?

**Review by kbottla — COMMENTED** [review]
Changes looks good. Added a few comments to cover testing the limits (max values) as the  changes(lowering) to `max` can go undetected.

File: src/main/java/uk/gov/pay/connector/charge/model/PrefilledCardHolderDetails.java (hunk 1 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/charge/model/PrefilledCardHolderDetails.java]: 👍

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiCreateResourceITest.java (hunk 3 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/resources/ChargesApiCreateResourceITest.java]: That should be max = 255, soz!

**Review by kbottla — APPROVED** [review]
Thanks for the changes, looks good to me 👍
